src/ucp/core/ucp_ep.inl

```diff
@@ -273,4 +273,12 @@ ucp_ep_config_connect_p2p(ucp_worker_h worker,
            !ucp_worker_is_tl_2iface(worker, rsc_index);
 }
 
+static UCS_F_ALWAYS_INLINE int ucp_ep_can_do_keepalive(ucp_ep_h ep)
+{
+    return (ep->cfg_index != UCP_WORKER_CFG_INDEX_NULL) &&
+           (ucp_ep_config(ep)->key.err_mode != UCP_ERR_HANDLING_MODE_NONE) &&
+           ucp_ep_config(ep)->key.ep_check_map &&
```

Comment:
should not `ep_check_map` be `0` if err handling is not enabled on configuration EP phase?  then `UCP_ERR_HANDLING_MODE_NONE ` check is redundant here

Reply:
check_ep_map is generated before err_mode is initialized & we don't know if it required or not

Reply:
how is it possible? IIUC check_ep_map has to be initialized on selection, err handling mode is defined at this time

Reply:
ok, fixed